compiler/rustc_codegen_llvm/src/type_of.rs

```diff
@@ -42,7 +43,12 @@ fn uncached_llvm_type<'a, 'tcx>(
         // FIXME(eddyb) producing readable type names for trait objects can result
         // in problematically distinct types due to HRTB and subtyping (see #47638).
         // ty::Dynamic(..) |
-        ty::Adt(..) | ty::Closure(..) | ty::Foreign(..) | ty::Generator(..) | ty::Str => {
+        ty::Adt(..) | ty::Closure(..) | ty::Foreign(..) | ty::Generator(..) | ty::Str
+            // For performance reason we use names only when emitting LLVM IR. Unless we are on
+            // LLVM < 14, where the use of unnamed types where involved in various issues, e.g.,
```

Comment:
```suggestion
            // LLVM < 14, where the use of unnamed types resulted in various issues, e.g.,
```
or something, not sure what "where involved" was supposed to be here.